apps/web/app/api/google/webhook/process-history-item.ts

```diff
@@ -292,3 +310,187 @@ export function shouldRunColdEmailBlocker(
     hasAiAccess
   );
 }
+
+async function handleLabelRemovedEvent(
+  message: gmail_v1.Schema$HistoryLabelRemoved,
+  {
+    gmail,
+    emailAccount,
+    provider,
+  }: {
+    gmail: gmail_v1.Gmail;
+    emailAccount: EmailAccountWithAI;
+    provider: EmailProvider;
+  },
+) {
+  const messageId = message.message?.id;
+  const threadId = message.message?.threadId;
+  const emailAccountId = emailAccount.id;
+  const userEmail = emailAccount.email;
+
+  if (!messageId || !threadId) {
+    logger.info("Skipping label removal - missing messageId or threadId", {
+      userEmail,
+      messageId,
+      threadId,
+    });
+    return;
+  }
+
+  const loggerOptions = {
+    email: userEmail,
+    messageId,
+    threadId,
+  };
+
+  logger.info("Processing label removal for learning", loggerOptions);
+
+  try {
+    const parsedMessage = await provider.getMessage(messageId);
+    const sender = extractEmailAddress(parsedMessage.headers.from);
+
+    // For label removal, we need to get the labelIds from the HistoryLabelRemoved object
+    const removedLabelIds = message.labelIds || [];
```

Comment:
note, we don't want to learn from INBOX label removed. that's just archiving emails

same for READ, and may also apply to things like SENT and DRAFT